curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md

```diff
@@ -514,3 +514,564 @@ Dynamically adjusting distribution and alignment of items for different viewport
+
+---
+
+It aligns flex items along the cross axis of the container.
+
+#### --answer--
+
+It determines if flex items should wrap onto new lines to fit available space.
+
+### --question--
+
+#### --text--
+
+What is the default behavior of the `flex-wrap: nowrap;` declaration?
+
+#### --distractors--
+
+It forces flex items to wrap onto a new line if they are too wide for the container.
+
+---
+
+It wraps items onto a new line, but in the reverse order.
+
+---
+
+It prevents the flex container from being larger than its parent element.
+
+#### --answer--
+
+It forces flex items to remain on a single line, even if they overflow the container.
+
+### --question--
+
+#### --text--
+
+Which CSS property is a shorthand for setting both the `flex-direction` and `flex-wrap` properties at once?
+
+#### --distractors--
+
+`flex-box`
+
+---
+
+`flex-align`
+
+---
+
+`flex-container`
+
+#### --answer--
+
+`flex-flow`
+
+### --question--
+
+#### --text--
+
+What is the effect of setting `flex-direction: column-reverse;` on a flex container?
+
+#### --distractors--
+
+It arranges items horizontally from right to left.
+
+---
+
+It reverses the items in a row, but keeps them on a single line.
+
+---
+
+It arranges items vertically from top to bottom.
+
+#### --answer--
+
+It arranges items vertically in a reverse order from bottom to top.
+
+### --question--
+
+#### --text--
+
+Which CSS snippet correctly arranges flex items in a vertical column and allows them to wrap onto a new column if they overflow the container's height?
+
+#### --distractors--
+
+```css
+.container {
+  display: flex;
+  flex-flow: row wrap;
+}
+```
+
+---
+
+```css
+.container {
+  display: flex;
+  flex-direction: column;
+  flex-wrap: nowrap;
+}
+```
+
+---
+
+```css
+.container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+```
+
+#### --answer--
+
+```css
+.container {
+  display: flex;
+  flex-flow: column wrap;
+}
+```
+
+### --question--
+
+#### --text--
+
+Which `justify-content` value aligns all flex items to the end of the container's main axis?
+
+#### --distractors--
+
+`justify-content: end;`
+
+---
+
+`justify-content: center;`
+
+---
+
+`justify-content: flex-start;`
+
+#### --answer--
+
+`justify-content: flex-end;`
+
+### --question--
+
+#### --text--
+
+What is the effect of the following CSS rule on flex items?
+
+```css
+.container {
+  flex-flow: row nowrap;
+}
+```
+
+#### --distractors--
+
+Items are arranged in a column and will wrap to a new column.
+
+---
+
+Items are arranged in a row and will wrap to a new row.
+
+---
+
+Items are arranged in a column and will overflow if they don't fit.
+
+#### --answer--
+
+Items are arranged in a row and will overflow the container if they don't fit.
+
+### --question--
+
+#### --text--
+
+Which CSS snippet will place the first flex item at the far left, the last item at the far right, and distribute the remaining items evenly in the space between them?
```

Comment:
this question seemed a little long to me.

It would be better to ask what the result would be for the following code snippet and then the answers would be describing the outcome